src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1335,19 +1335,25 @@ private Symbol UnwrapAlias(Symbol symbol, out AliasSymbol alias, BindingDiagnost
         private ImmutableArray<TypeWithAnnotations> BindTypeArguments(SeparatedSyntaxList<TypeSyntax> typeArguments, BindingDiagnosticBag diagnostics, ConsList<TypeSymbol> basesBeingResolved = null)
         {
             Debug.Assert(typeArguments.Count > 0);
-            var args = ArrayBuilder<TypeWithAnnotations>.GetInstance();
+            var args = ArrayBuilder<TypeWithAnnotations>.GetInstance(typeArguments.Count);
             foreach (var argSyntax in typeArguments)
-            {
                 args.Add(BindTypeArgument(argSyntax, diagnostics, basesBeingResolved));
-            }
 
             return args.ToImmutableAndFree();
         }
 
         private TypeWithAnnotations BindTypeArgument(TypeSyntax typeArgument, BindingDiagnosticBag diagnostics, ConsList<TypeSymbol> basesBeingResolved = null)
         {
-            // Unsafe types can never be type arguments, but there's a special error code for that.
-            var binder = this.WithAdditionalFlags(BinderFlags.SuppressUnsafeDiagnostics);
+            // BackCompat.  The compiler would previously suppress reporting errors for pointers in generic types.  This
+            // was intended so you would get a specific error in CheckBasicConstraints.CheckBasicConstraints for a type
+            // like (like `List<int*>`). i.e. you would get the error about an unsafe type not being a legal type argument,
+            // but not the error about not being in an unsafe context.  This had the unfortunate consequence though of 
+            // preventing the latter check for something like `List<int*[]>`.  Here, this is a legal generic type, but we 
+            // still want to report the error that you need to be in an unsafe context.  So, to maintain compat, we only
+            // do the suppression if you're on C# 11 and prior.  In later versions we do the correct check.
+            var binder = ((CSharpParseOptions)typeArgument.SyntaxTree.Options).LanguageVersion < LanguageVersion.Preview
```

Comment:
> ((CSharpParseOptions)typeArgument.SyntaxTree.Options).LanguageVersion < LanguageVersion.Preview [](http://example.com/codeflow?start=24&length=96)

It looks like this is going to change what error we report for pointer types at the top level of the type argument. That is the likely source of the tests failures. It doesn't feel like a change like that would be desired

Reply:
> It doesn't feel like a change like that would be desired

Depending on the actual effect

Reply:
It feels appropriate to me based on the failures so far.  You get a good error if `unsafe` isn't present... which i think is totally reasonable for that situation.  It's telling you the two very relevant details:

1. you need to be in an unsafe context to use pointers.
2. your pointers aren't legal here because you're using generics.  

Personally, i htink it's a good change :)